### PR TITLE
Making FormPart generic

### LINKED ISSUE #5801: Form-FlowPart template should be more generic
FormPart Description says: 

> Turns your content item into a form.

If you add the form content part to your custom content type it won't render a form because the https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Forms/Views/Form-FlowPart.cshtml template is connected to the _Form_ content type.

### PR DESCRIPTION
**Pull request by ns8482e**
Fixes #5801 

- Create `<form>` tag around `Content Type` that contains `FormPart`
-  Removed reference to Flow Module
- Removed FlowPart template customization from Form module
- Added support for multipart/form-data https://github.com/OrchardCMS/OrchardCore/issues/6027#issuecomment-619504342  by @dodyg 

image

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Forms/Views/Form.Wrapper.cshtml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+@inject OrchardCore.ContentManagement.Display.IContentItemDisplayManager ContentItemDisplayManager
+
+@{
+    var formPart = ((ContentItem)Model.ContentItem).As<FormPart>();
+    var widgets = (JArray)((JObject)Model.ContentItem.Content)["FlowPart"]["Widgets"];
+    bool hasFileInput = false;
+
+    if(widgets.Count() > 0 )
+    {
+        hasFileInput = widgets?.Where( a=> a["InputPart"] != null && a["InputPart"]["Type"]?.Value<string>() == "file" )?.Count() > 0;
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Forms/Views/Form.Wrapper.cshtml:15]: This should be an option. Either as settings or the part.

File: src/OrchardCore.Modules/OrchardCore.Forms/Views/Form.Wrapper.cshtml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    if(widgets.Count() > 0 )
+    {
+        hasFileInput = widgets?.Where( a=> a["InputPart"] != null && a["InputPart"]["Type"]?.Value<string>() == "file" )?.Count() > 0;   
+    }
+
+    string encType = hasFileInput ? "multipart/form-data" : "application/x-www-form-urlencoded";  
+}
+
+<form action="@formPart.Action" method="@formPart.Method"  enctype="@encType"  >
+    @Html.AntiForgeryToken()
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Forms/Views/Form.Wrapper.cshtml:22]: This should be an option. Either as settings or the part.

File: src/OrchardCore.Modules/OrchardCore.Forms/Views/Form.Wrapper.cshtml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    var formPart = ((ContentItem)Model.ContentItem).As<FormPart>();
+    var widgets = (JArray)((JObject)Model.ContentItem.Content)["FlowPart"]["Widgets"];
+    bool hasFileInput = false;
+
+    if(widgets.Count() > 0 )
+    {
+        hasFileInput = widgets?.Where( a=> a["InputPart"] != null && a["InputPart"]["Type"]?.Value<string>() == "file" )?.Count() > 0;   
+    }
+
+    string encType = hasFileInput ? "multipart/form-data" : "application/x-www-form-urlencoded";
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Forms/Views/Form.Wrapper.cshtml:18]: Should it be an option on the Form part? With a drop-down of predefined values?

Comment (ns8482e) [reply src/OrchardCore.Modules/OrchardCore.Forms/Views/Form.Wrapper.cshtml:18]: I thought to make it less technical assign automatically based on widgets added

File: src/OrchardCore.Modules/OrchardCore.Forms/Views/Form.Wrapper.cshtml (hunk 1 above)
Comment (ns8482e) [reply src/OrchardCore.Modules/OrchardCore.Forms/Views/Form.Wrapper.cshtml]: moved to part

File: src/OrchardCore.Modules/OrchardCore.Forms/Views/Form.Wrapper.cshtml (hunk 2 above)
Comment (ns8482e) [reply src/OrchardCore.Modules/OrchardCore.Forms/Views/Form.Wrapper.cshtml]: moved to part

File: src/OrchardCore.Modules/OrchardCore.Forms/Views/Form.Wrapper.cshtml (hunk 3 above)
Comment (ns8482e) [reply src/OrchardCore.Modules/OrchardCore.Forms/Views/Form.Wrapper.cshtml]: moved to part

File: src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormContentDisplayDriver.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+using OrchardCore.Forms.Models;
+
+namespace OrchardCore.Forms.Drivers
+{
+    public class FormContentDisplayDriver : ContentDisplayDriver
+    {
+        public override Task<IDisplayResult> DisplayAsync(ContentItem model, BuildDisplayContext context)
+        {
+            var formItemShape = context.Shape;
+            //If content item contains FormPart add Form Wrapper.
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormContentDisplayDriver.cs:16]:
```suggestion
            // If the content item contains FormPart add Form Wrapper.
```

File: src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormContentDisplayDriver.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+            var formPart = model.As<FormPart>();
+            if (formPart != null)
+            {
+                formItemShape.Metadata.Wrappers.Add($"Form_Wrapper");
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormContentDisplayDriver.cs:20]:
```suggestion
                formItemShape.Metadata.Wrappers.Add("Form_Wrapper");
```

File: src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormContentDisplayDriver.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+            }
+
+            //We don't need to return a shape result
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormContentDisplayDriver.cs:23]:
```suggestion
            // We don't need to return a shape result
```

File: src/OrchardCore.Modules/OrchardCore.Forms/Views/Items/FormPart.Fields.Edit.cshtml [hunk 7]
```diff
@@ -16,6 +16,24 @@
...
+        <option value="text/plain">text/plain</option>
+    </select>
+    <span class="hint">@T["The enctype to use when submitting the form using POST method."]</span>
+</div>
+
+<div class="form-group pb-1">
+    <div class="custom-control custom-checkbox">
+        <input asp-for="EnableAntiForgeryToken" class="custom-control-input" />
+        <label asp-for="EnableAntiForgeryToken" class="custom-control-label">@T["Anti Forgery Token"]</label>
+        <span class="hint">@T["Enable or Disable Anti Forgery Token."]</span>
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Forms/Views/Items/FormPart.Fields.Edit.cshtml:33]:
```suggestion
        <span class="hint">@T["Check to generate an Anti Forgery Token in the form."]</span>
```

**Comment by jwu-au** [comment]
Hi @ns8482e , do we want to make below as default for this PR?
```
{
  "TitlePart": [
    {
      // hides title part for Form and label type
      "displayType": "Detail",
      "contentType": ["LiquidPage", "Form", "Label"],
      "place": "-"
    }
  ]
}
```

**Comment by ns8482e** [comment]
@jwu-au I guess TitlePart has setting to hide it, isn't it?

**Pull request merged**